integration_tests/src/main/python/cast_test.py

```diff
@@ -667,14 +667,15 @@ def test_cast_int_to_string_not_UTC():
         {"spark.sql.session.timeZone": "+08"})
 
 not_utc_fallback_test_params = [(timestamp_gen, 'STRING'), (timestamp_gen, 'DATE'),
-        (date_gen, 'TIMESTAMP'),
+        # python does not like year 0, and with time zones the default start date can become year 0 :(
+        (DateGen(start=date(2, 1, 1)), 'TIMESTAMP'),
```

Comment:
Nit: Should this be `date(1, 1, 2)` so we don't skip almost an entire year of candidate test dates?

Reply:
I was being overly cautious and gave it a year, but yes the test is only 8 hours off from UTC, so it should just be a day.

Reply:
I had to go with `date(1, 1, 3)` to not fail with seed 4. Not totally sure why a +8 zone would cause that, but it is CPU only so I will just trust it.